### DISCUSSION TITLE
Policy suggestion: Use rules_testing for new tests on rules

### DISCUSSION THREAD

**Discussion by matts1** [bazelbuild/rules_rust#2550] [Ideas]
I notice that we've been using `@bazel_skylib`'s analysis_tests (eg. [here](https://github.com/bazelbuild/rules_rust/blob/4aafa0af925b855432020e197e474bb180adf9e4/test/toolchain/toolchain_test.bzl)).

I've recently been using [`rules_testing`](https://rules-testing.readthedocs.io/en/latest/analysis_tests.html) to add a bunch of tests to `rules_cc`, and I've found the experience to be much better.

Do we want to make a policy that new tests for rules use `rules_testing`?

**Comment by tidefield** [comment]
I wrote [a few tests](https://github.com/search?q=repo%3Abazelbuild%2Frules_rust%20rules_testing&type=code) with rules_testing and agree that it's a much better experience and saves a lot of boilerplate code.

I already introduced it in https://github.com/bazelbuild/rules_rust/pull/2480. @UebelAndre thought it's better idea to convert a few existing tests to get a better sense of the improvements in comparison before we enforce it.

I haven't got the time to do so.

**Reply by matts1** [reply]
Ah, I didn't notice that, that's awesome!

FWIW, I've been working with it quite a lot recently, and I've found that while it does have some things with it that I take issue with, even with those issues it's still a much better than the alternative.

In general, I'd say that it works very well for simple types, but for more complex types it lacks proper generics (see https://github.com/bazelbuild/rules_testing/issues/93)
